test/inductor/test_efficient_conv_bn_eval.py

```diff
@@ -0,0 +1,166 @@
+
+importlib.import_module("functorch")
+importlib.import_module("filelock")
+
+from inductor.test_torchinductor import copy_tests
+
+
+class ConvOp(nn.Module):
+    expected_optimization_count = 1
+
+    def __init__(
+        self,
+        conv_class,
+        bn_class,
+        use_bias,
+        in_channels,
+        out_channels,
+        device,
+        **kwargs,
+    ):
+        super().__init__()
+        self.conv = conv_class(in_channels, out_channels, bias=use_bias, **kwargs).to(
+            device
+        )
+        self.bn = bn_class(out_channels).to(device)
+
+    def forward(self, x):
+        x = self.conv(x)
+        return self.bn(x)
+
+
+class MultiUserConvOp(nn.Module):
+    expected_optimization_count = 3
+
+    def __init__(
+        self,
+        conv_class,
+        bn_class,
+        use_bias,
+        in_channels,
+        out_channels,
+        device,
+        **kwargs,
+    ):
+        super().__init__()
+        self.conv1 = conv_class(in_channels, out_channels, bias=use_bias, **kwargs).to(
+            device
+        )
+        self.bn1 = bn_class(out_channels).to(device)
+        self.conv2 = conv_class(out_channels, out_channels, bias=use_bias, **kwargs).to(
+            device
+        )
+        self.bn2 = bn_class(out_channels).to(device)
+        self.conv3 = conv_class(out_channels, out_channels, bias=use_bias, **kwargs).to(
+            device
+        )
+        self.bn3 = bn_class(out_channels).to(device)
+
+    def forward(self, x):
+        # this conv-bn pair can use efficient_conv_bn_eval
+        x = self.bn1(self.conv1(x))
+        # this conv-bn pair cannot use efficient_conv_bn_eval feature
+        # just for the second forward of the `self.conv2`
+        x = self.bn2(self.conv2(self.conv2(x)))
+        # this conv-bn pair can use efficient_conv_bn_eval feature
+        # just for the first forward of the `self.bn3`
+        x = self.bn3(self.bn3(self.conv3(x)))
+        return x
+
+
+class EfficientConvBNEvalTemplate(TestCase):
+    def test_basic(self):
+        def test_conv_bn_eval(test_class, use_bias, module):
+            mod_eager = test_class(
+                module[0],
+                module[1],
+                use_bias,
+                3,
+                32,
+                self.device,
+                kernel_size=3,
+                stride=2,
+            ).eval()
+            torch._dynamo.reset()
+            mod_optimized = torch.compile(mod_eager)
+
+            inps = [4, 3, 96]
+            if module[0] == nn.Conv2d:
+                inps.append(inps[-1])
+            if module[0] == nn.Conv3d:
+                inps.append(inps[-1])
+                inps.append(inps[-1])
+
+            original_value = counters["inductor"]["efficient_conv_bn_eval"]
+            inp = torch.rand(inps).to(self.device)
+            out_eager = mod_eager(inp)
+            out_optimized = mod_optimized(inp)
+            self.assertEqual(out_optimized, out_eager, atol=2e-04, rtol=1e-5)
```

Comment:
This is numerics changing. something to watch out for. I'm going to add a benchmarking run to see if we get any accuracy failures.

Reply:
What do you mean by that? I follow the test from [test/inductor/test_binary_folding.py](https://github.com/pytorch/pytorch/pull/108757/files/604ce2f6eaa764674210572eef81f61f19980c6d#diff-32177bda568814a1368e62144d5b08ad18bb92f5578d9994a5350c516d5e0294).